extension.js

```diff
@@ -141,8 +170,57 @@ class PipOnTop
       /* Change stick if enabled or unstick PipAble windows */
       un = (isPipWin && this.settings.get_boolean('stick')) ? '' : 'un';
       window[`${un}stick`]();
+
+      /* Repeatedly override new window position so it sticks */
+      window._overrideTimeoutId = GLib.timeout_add(GLib.PRIORITY_DEFAULT, 1000, () => {
+        window._overrideTimeoutId = null;
+        return false;
+      });
```

Comment:
This seems super hacky, what exactly you are working around here? Not a big fan of these kind of solutions, maybe we can think of something better if possible.

Reply:
Without interfering, Firefox pip windows make 3 position and 2 resize changes. With the extension active that increases to 4 position and 3 resize changes. If a pip window overlaps any other "Always on top" window (which is what happens when opening multiple pip windows) it makes a few hundred attempts before triggering a recursion error. Limiting the number of override attempts causes the window to shrink but this can be corrected by applying the size again at the end of the timeout. I suspect this might have something to do with tiling which can be seen on X11.